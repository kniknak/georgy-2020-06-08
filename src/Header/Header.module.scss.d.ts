declare const styles: {
    readonly root: string;
    readonly search: string;
    readonly upload: string;
};
export = styles;