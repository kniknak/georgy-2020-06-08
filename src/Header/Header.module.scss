@mixin mobile {
    @media only screen and (max-width: 375px) {
        @content;
    }
}

.root {
    display: flex;
    justify-content: space-between;
    padding: 12px;
    margin-bottom: 64px;

    @include mobile {
        flex-wrap: wrap;
    }
}

.search {
    max-width: 500px;
    width: 500px;
    margin-right: 12px;

    @include mobile {
        width: 100%;
        margin-right: 0;
        margin-bottom: 12px;
    }
}

.upload {
    max-width: 200px;
    width: 200px;

    @include mobile {
        max-width: 100%;
        width: 100%;
    }
}