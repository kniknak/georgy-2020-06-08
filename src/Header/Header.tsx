import React from "react";
import { FileUpload } from "./FileUpload/FileUpload";
import { Search } from "./Search/Search";
import * as styles from "./Header.module.scss";

export interface HeaderProps {
    addFile: (file: File) => void
    search: string
    setSearch: (search: string) => void
}

export const Header: React.FC<HeaderProps> = ({ addFile, search, setSearch }) => {
    return (
        <div className={styles.root}>
            <div className={styles.search}>
                <Search
                    search={search}
                    setSearch={setSearch}
                />
            </div>
            <div className={styles.upload}>
                <FileUpload addFile={addFile}/>
            </div>
        </div>
    );
};
