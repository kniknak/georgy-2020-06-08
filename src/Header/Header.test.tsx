import React from "react";
import { shallow } from "enzyme";
import { Header, HeaderProps } from "./Header";

describe("Header", () => {
    it("renders", () => {
        const props: HeaderProps = {
            addFile: () => undefined,
            setSearch: () => undefined,
            search: "foo",
        }
        const subject = shallow(<Header {...props}/>);

        expect(subject).toMatchSnapshot();
    });
});