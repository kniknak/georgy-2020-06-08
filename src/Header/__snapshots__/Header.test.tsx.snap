// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<div>
  <div>
    <Search
      search="foo"
      setSearch={[Function]}
    />
  </div>
  <div>
    <FileUpload
      addFile={[Function]}
    />
  </div>
</div>
`;
